monai/networks/layers/simplelayers.py

```diff
@@ -434,6 +435,109 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         return torch.as_tensor(ht, device=ht.device, dtype=ht.dtype)
 
 
+def get_binary_kernel3d(window_size: Sequence[int]) -> torch.Tensor:
+    r"""Create a binary kernel to extract the patches.
+    If the window size is HxWxD will create a (H*W*D)xHxWxD kernel.
+    """
+    window_range: int = window_size[0] * window_size[1] * window_size[2]
+    kernel: torch.Tensor = torch.zeros(window_range, 1, window_size[0], window_size[1], window_size[2])
+    for i in range(window_size[0]):
+        for j in range(window_size[1]):
+            for k in range(window_size[2]):
+                l: int = i * window_size[2] * window_size[1] + j * window_size[2] + k
+                kernel[l, 0, i, j, k] += 1.0
+    return kernel
+
+
+def _compute_zero_padding(kernel_size: Sequence[int]) -> List[int]:
+    r"""Utility function that computes zero padding tuple."""
+    computed: List[int] = [(k - 1) // 2 for k in kernel_size]
+    return computed
+
+
+def median_filter(in_tensor: torch.Tensor, kernel_size: Sequence[int], kernel: torch.Tensor = None) -> torch.Tensor:
+    r"""Apply median filter to an image.
+    Args:
+        in_tensor: the input image with shape :math:`(C,H,W,D)`.
+        kernel_size: the convolution kernel size.
+    Returns:
+        the filtered input tensor with shape :math:`(C,H,W,D)`.
+    Example:
+        >>> x = torch.rand(4, 5, 7, 6)
+        >>> output = median_filter(x, (3, 3, 3))
+        >>> output.shape
+        torch.Size([4, 5, 7, 6])
+    """
+    if not isinstance(in_tensor, torch.Tensor):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(in_tensor)}")
+
+    original_shape = in_tensor.shape
+    if len(original_shape) == 5 and original_shape[0] == 1:
+        in_tensor = in_tensor.squeeze(0)
+    if not len(in_tensor.shape) == 4:
+        raise ValueError(f"Invalid in_tensor shape, we expect CxHxWxD. Got: {in_tensor.shape}")
+
+    padding: Sequence[int] = _compute_zero_padding(kernel_size)
+
+    # prepare kernel
+    if kernel is None:
+        kernel: torch.Tensor = get_binary_kernel3d(kernel_size).to(in_tensor)
+    else:
+        kernel: torch.Tensor = kernel.to(in_tensor)
+    c, h, w, d = in_tensor.shape
+
+    # map the local window to single vector
+    features: torch.Tensor = F.conv3d(in_tensor.reshape(c, 1, h, w, d), kernel, padding=padding, stride=1)
+    features = features.view(c, -1, h, w, d)  # Cx(K_h * K_w * K_d)xHxWxD
```

Comment:
```suggestion
        kernel: torch.Tensor = get_binary_kernel(kernel_size).to(in_tensor)
    else:
        kernel: torch.Tensor = kernel.to(in_tensor)
    c, *sshape = in_tensor.shape

    # map the local window to single vector
    features: torch.Tensor = F.conv3d(in_tensor.reshape(c, 1, *sshape), kernel, padding=padding, stride=1)
    features = features.view(c, -1, *sshape)  # Cx(K_h * K_w * K_d)xHxWxD
```
I think we can get this to work in 2D as well.

Reply:
The tests pass with this change.